SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/PredictedGameplayCuesTest/PredictedGameplayCuesTest.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "CuesGASTestPawn.h"
+#include "SpatialFunctionalTest.h"
+#include "PredictedGameplayCuesTest.generated.h"
+
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API APredictedGameplayCuesTest : public ASpatialFunctionalTest
+{
+	GENERATED_BODY()
+
+	APredictedGameplayCuesTest();
+
+	virtual void PrepareTest() override;
+
+	UPROPERTY(EditInstanceOnly, Category = "Test Settings")
+	float DuplicateActivationCheckWaitTime;
+
+private:
+	ACuesGASTestPawn* TestPawn;
+	APawn* PrevPawn;
+	float StepTimer;
```

Comment:
Should have thought of this earlier, as a bit late now, but I am pretty sure unreal has some Timer classes you can use for next time.

Reply:
Mhh yea hadn't considered that. The only timer funtionality I've found just now is the FTimerManager, which registers a callback. I think using that wouldn't clean this up much more though, since you'd still have to track whether you started a timer.
If this becomes a pattern in tests it'll be worth trying to generalise.